### ISSUE TITLE
All NaN weights for feature 0

### ISSUE THREAD

**Issue by huisuncadd** [LPDI-EPFL/masif-neosurf#28]
Hi! 

Thanks for making MaSIF-neosurf publicly available. The study is impressive, and I am trying to apply the tool to my research.

I have encountered all NaN values in the output descriptor files (xxx.npy) and traced the issue to **all-NaN weights for feature 0** at masif/data/masif_ppi_search/nn_models/sc05/all_feat/model_data/model.

I would greatly appreciate any guidance on how to resolve this issue.

I look forward to hearing from you.

Best,
Huisun

**Comment by huisuncadd** [comment]
I also share my additional analysis results on NaN weights for feature 0 at masif/data/masif_ppi_search/nn_models/sc05/all_feat/model_data/model.

When I installed MaSIF-neosurf using Docker, it successfully completed the surface descriptor calculations (feauture_A.npy).
I also calculated descriptors for the identical input structures after replacing all NaN values with 0 (feauture_B.npy).
When I compared feauture_A.npy with feauture_B.npy, both were exactly the same. It means that
- **Old tensorflow version (e.g., TF 1.12) silently treats NaN as non-positive (clamping to 0)**, which accidentally masks the broken W_conv_0. 
- On the other hand, the recent tensor flow version (e.g., TF 2.12) will correctly propagate NaN, exposing both the broken weights and the broken preprocessing.

**Comment by arneschneuing** [comment]
Hi Huisun, we already discussed this via email but I also wanted to leave a comment here for everyone else. We had also encountered this issue a while ago and discovered that it was caused by NaN values in a few examples from the original MaSIF training set. The problem didn't surface earlier because TensorFlow 1.12 indeed silently masks these values. We already fixed the problem with [this commit](https://github.com/LPDI-EPFL/masif-neosurf/commit/65e3e515587f3e20ed15178b01d2d29b2cdd792b#diff-958558be40a79090125054dd1b54730da089d6ea826715b88c16458bb4799393L194) but decided to publish the original "broken" weights because they are compatible with the pre-computed databases. As long as you use the provided inference container (or at least an environment with TensorFlow 1.12) this shouldn't cause any downstream problems.
